### ISSUE TITLE
IPX network number isn't taken into account?

### ISSUE THREAD

**Issue by superfury** [jsalin/ipxgw#12]
I notice that the IPX network number isn't taken into account when sending packets or receiving packets?

Afaik, IPX network number 0 in a packet is special (denoting 'current network number'), so it might need to be replaced with the client's own network number before sending it out on the wire? It also isn't routable. Once on the wire, any connected client can't know what network it originated from in this case (as all clients would give '0').

Perhaps replace the network number in the source and destination fields with the client's for the fields being 0?

**Comment by superfury** [comment]
OK. Made a basic filter on the IPX network number source/destination fields (trying to base it off what's already implemented with the destination address). Got any explanation what that source node number filter is for in the broadcast address (it looked like it was saying: not IP unequal or not port unequal? I've currently applied it to just be performed when the source network is 0 combined with the node pointing to the assigned address being filtered out?

It seems to run fine with Dosbox clients so far, even running split IPX networks properly (and they don't talk to each other's network because of that!). But when I add UniPCemu's packet server client to the mix, it will seem to fail now and not proceed into the Doom 95+Dosbox MS-DOS DOOM IPX game (it seems to be stuck inside the discovery-ish phase, detecting both clients with a spinning star-shaped sign (using /, \ and related characters to make it spin) next to player 1 infinitely)?
Edit: OK. So the app properly sends/receives packets it seems, but my own emulator doesn't seem to be receiving anything with it's pcap interface. Perhaps a case of missing application rights.

**Comment by superfury** [comment]
Managed to get the entire system (multiplayer with and without players on the real IPX network (Ethernet II) and/or UniPCemu's clients (which are counted as the same using this application's code)) working properly.

I've also implemented a working filter of incoming IPX packets' network number. So when a packet is received for destination network number 0, the source network is used instead. Also source and destination network numbers having all bits set are counted as a broadcast network (which effectively ignores the network number in the filters). The source network number can be non-'all bits set' (use the specified network number directly) or set to all-ones (FFFFFFFFh), counting as a broadcast.

The code I've written takes all those cases into account.

I've even ran simulataneous games on the same port number with different network numbers, and it ran without issues, showing that the network number is functional and working properly!

This is now implemented in pull request #13
